### PR TITLE
Add error message when email is nil or of a different pattern

### LINKED ISSUE #5397: Display error message in  the edit profile form if the email is nil or does matches the email pattern
When we enter the blank email, there was no error message saying that the email field is nil and is required. Also, no error message when the email pattern is not matched. 
Moreover, when the above two requirements fail, the changes made are not saved and the "save" button gets disabled and gets replaced by a fa-spinner, due to which a user has to reload a page to make the changes again. 

Screen Shot 2019-04-07 at 4 33 00 PM

### PR DESCRIPTION
**Pull request by gautamig54**
Fixes #5397  (<=== Add issue number here)

Adds an error message to the text field when the email field is blank or of a different pattern. 
Also, prevents the save button to be disabled when the email field entered is invalid. 

Screen Shot 2019-04-07 at 4 34 47 PM

Screen Shot 2019-04-07 at 4 35 01 PM

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by gautamig54** [comment]
@publiclab/reviewers Kindly review this PR. Thank you!

**Comment by CleverFool77** [comment]
Hi @gautamig54 ,
I did something similar in https://github.com/publiclab/plots2/pull/5368 but didn't test with regex though.
Instead use jquery-validation
Great Work !!!

**Comment by CleverFool77** [comment]
Hi @gautamig54 ,
I did something similar in https://github.com/publiclab/plots2/pull/5368 but didn't test with regex though.
Instead use jquery-validation

and @ananyaarun worked on https://github.com/publiclab/plots2/pull/5116,
I guess all our PRs are based on validation.
Great Work !!!

**Comment by gautamig54** [comment]
@CleverFool77 Yes. All our PRs are based on validation. Great work in #5368!!!
I have added validation for emails since you did it for passwords!

**Comment by CleverFool77** [comment]
Hi @gautamig54 
Actually I added validation for all form fields but used jquery-validation instead of regex.

**Comment by gauravano** [comment]
If jquery-validation is added in any PR then I think we can use same at other places too. Although, I think it's fine here as we are just putting on Email? 

Does anyone want to highlight the advantages of using a package? Also, is this ready?

**Comment by gautamig54** [comment]
@gauravano Yes, this PR is ready and working fine!

File: app/views/users/_edit_form.html.erb [hunk 1]
```diff
@@ -75,8 +75,9 @@
       <script>
         (function() {
           $('.btn-save').click(function onClick(e) {
-            $(this).addClass("disabled") // disable the button after it is clicked
-                   .html("<i class='fa fa-spinner fa-spin'></i>"); // make a spinner that spins when clicked
+            if(document.form.email.text_field != "" || document.form.email.pattern == '[^@]+@[^@]+\.[a-zA-Z]{2,6}')
+                   {   $(this).addClass("disabled") // disable the button after it is clicked
```
Comment (gauravano) [inline app/views/users/_edit_form.html.erb]: @gautamig54 could you do a little indentation and formatting here? Thanks!

**Comment by gauravano** [comment]
Merged :tada: :balloon: :100:

**Pull request merged**